File: addons/proxy/src/main/java/dev/httpmarco/polocloud/addons/proxy/ProxyConfig.java
package dev.httpmarco.polocloud.addons.proxy;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.experimental.Accessors;

@Getter
@Accessors(fluent = true)
@AllArgsConstructor
public final class ProxyConfig {

    private final Motd[] motds;
    private final Motd[] maintenanceMotds;
    private final String maintenancePlayerInfo;
    private final Tablist tablist;

}

File: addons/proxy/src/main/kotlin/dev/httpmarco/polocloud/addons/proxy/ProxyConfig.kt
package dev.httpmarco.polocloud.addons.proxy

import dev.httpmarco.polocloud.addons.proxy.motd.Motd
import dev.httpmarco.polocloud.addons.proxy.tablist.Tablist

interface ProxyConfig {
    fun prefix(): String
    fun messages(key: String): String
    fun aliases(): List<String>
    fun tablist(): Tablist
    fun motd(): Motd
    fun maintenanceMotd(): Motd
}